ActionButton.js

```diff
@@ -0,0 +1,30 @@
+import { Button } from "./Button.js";
+
+export class ActionButton extends Button {
+  constructor(sign, calculator, buttonContainer) {
+    super(sign, calculator, buttonContainer);
+    this.initializeEventListener();
+  }
+  initializeEventListener = () => {
+    this.buttonContainer.addEventListener("click", this.clickOnAction);
+  };
+  clickOnAction = () => {
+    this.actionButtonPressed(this.sign);
+  };
+  actionButtonPressed = (sign) => {
+    if (this.calculator.input.minusClickedBeforeNumber(sign)) {
+      this.calculator.input.inputContainer.innerHTML = "-";
+    }
+    if (this.calculator.input.actionButtonPressedWhileFirstOperation(sign)) {
+      this.calculator.input.saveFirstNumberAndSign(sign);
+    } else if (this.calculator.input.eqButtonPressedWhileFirstOperation(sign)) {
+    } else if (this.calculator.input.eqButtonPressedWithNoInput(sign)) {
```

Comment:
Why do you need those else if statements? 

Maybe create an early exit condition above?
```
if(this.calculator.input.eqButtonPressedWhileFirstOperation(sign)
|| this.calculator.input.eqButtonPressedWithNoInput(sign)) {
  return;
}
```